src/script/interpreter.cpp

```diff
@@ -289,16 +289,51 @@ namespace {
  * expose whether the stack is empty and whether or not any false values are
  * present at all. To implement OP_ELSE, a toggle_top modifier is added, which
  * flips the last value without returning it.
+ *
+ * This uses an optimized implementation that does not materialize the
+ * actual stack. Instead, it just stores the size of the would-be stack,
+ * and the position of the first false value in it.
  */
 class ConditionStack {
 private:
-    std::vector<bool> m_flags;
+    //! The size of the implied stack.
+    uin32_t m_stack_size = 0;
+    //! The position of the first false value on the implied stack, or -1 if all true.
+    int32_t m_first_false_pos = -1;
```

Comment:
Using `int32_t` gives "comparison of integers of different signs" warnings. Having `uint32_t m_first_false_pos_plus_one = 0` instead avoids that, and seems to make the code simpler (avoids a bunch of decrements/increments). I think you could call it `m_size_of_stack_including_first_false` with 0 as a special value, and have it make sense, maybe?

Reply:
I'm instead adding a NO_FALSE constant (equal to 0xFFFFFFFF), and reordering operations a bit to avoid some `- 1`s.